### PR TITLE
Refactor: Context Keys

### PR DESCRIPTION
**Pull request by glennsl**
This creates a new context keys data structure and schema, so that context keys can be defined as contributions and extended with context-specific context keys (:roll_eyes:).

### CODE REVIEW

File: src/Model/ContextKeys.re [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        fun
+        | Some({variant: EditorsPicker, _}) => true
+        | _ => false,
+      ),
+    ],
+  );
+
+let editors =
+  fromList([
+    bool("insertMode", state =>
```
Comment (bryphe) [inline src/Model/ContextKeys.re:29]: Much nicer than what we had before!

File: src/Core/WhenExpr/ContextKeys.re [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  // Emulate JavaScript semantics
+  let asBool =
+    fun
+    | True => true
+    | False => false
+    | String("") => false
+    | String(_) => true;
+
+  // Emulate JavaScript semantics
+  let asString =
```
Comment (bryphe) [inline src/Core/WhenExpr/ContextKeys.re:19]: Cool, we support string context now!

File: src/Core/WhenExpr/WhenExpr.rei [hunk 3]
```diff
@@ -9,6 +9,30 @@ module Value: {
   let asString: t => string;
 };
 
+module ContextKeys: {
+  module Schema: {
```
Comment (bryphe) [inline src/Core/WhenExpr/WhenExpr.rei:13]: Great API!

File: src/Core/WhenExpr/ContextKeys.re (hunk 2 above)
Comment (bryphe) [reply src/Core/WhenExpr/ContextKeys.re:19]: Is there any context key we're using this now? Or is this in preparation for the additional SCM functionality?

Comment (glennsl) [reply src/Core/WhenExpr/ContextKeys.re:19]: This was added when I revamped `WhenExpr`. It's just been moved here. It's currently just used in unit tests, but will be needed for the SCM functionality shortly.

**Pull request merged**